src/app/tasks/task.service.ts

```diff
@@ -59,15 +50,12 @@ export class TaskService {
         return basicTasks;
     }
 
-    public getTasks() : {tasks: BasicTask[]} {
-        // Sort tasks alphabetically
-        return {
-            tasks: this.convertTasksToBasicTasks(this.tasks).slice()
-            .sort(
-                (taskA, taskB) => {
-                    return (taskA.title < taskB.title) ? -1 : (taskA.title > taskB.title) ? 1 : 0
-                }
-            )};
+    public getTasks() : BasicTask[] {
+        return this.convertTasksToBasicTasks(this.tasks).slice()
```

Comment:
instead of using a variable here, you can call the endpoint which fetches tasks modify the incoming data however you like and save the state of data inside an observable and behaviorsubject (applies for taskTypes and taskStatuses as well)

Reply:
That way the state will be reactively updated in case any changes happen on the site, let's say you delete a task, you can then return a list of new tasks without the deleted task from the DB, pass the tasks to the tasks behaviorsubject and you have successfully updated the state of tasks

Reply:
if you need help with this I can explain this over call